python/venv/private/venv_common.bzl

```diff
@@ -458,28 +459,20 @@ def _create_python_zip_file(
     args.add("--main", _rlocationpath(main, ctx.workspace_name))
     args.add("--py_runtime", _rlocationpath(interpreter, ctx.workspace_name))
     args.add("--venv_process_wrapper", _rlocationpath(venv_toolchain.process_wrapper, ctx.workspace_name))
-    if venv_toolchain.zipapp_shebang:
-        args.add("--shebang", venv_toolchain.zipapp_main)
+    optional_shebang = shebang or venv_toolchain.zipapp_shebang
+    if optional_shebang:
+        args.add("--shebang", optional_shebang)
     args.add("--output", python_zip_file)
     args.add("--venv_config_info", json.encode(venv_config_info))
-
-    workspace_name = ctx.workspace_name
-
-    def _runfiles_filter_map(file):
-        return "{}={}".format(file.path, _rlocationpath(file, workspace_name))
-
-    args.add_all(venv_runfiles.files, map_each = _runfiles_filter_map, format_each = "--rfile=%s", allow_closure = True)
-
-    args.use_param_file("@%s", use_always = True)
-    args.set_param_file_format("multiline")
+    args.add("--runfiles_manifest", files_to_run.runfiles_manifest)
 
     ctx.actions.run(
         mnemonic = "PyVenvZipapp",
         executable = interpreter,
         arguments = [python_args, args],
         outputs = [python_zip_file],
-        inputs = depset([main, venv_toolchain.zipapp_main], transitive = [venv_runfiles.files]),
-        tools = depset([venv_toolchain.zipapp_maker], transitive = [py_runtime.files]),
+        inputs = venv_runfiles,
+        tools = [venv_toolchain.zipapp_maker],
         env = ctx.configuration.default_shell_env,
         resource_set = _zip_resource_set,
```

Comment:
Hm yeah per the docs this should be set, but can't say I perfectly why this would not correctly use exec without the arg.
> If executable and tools are coming from a toolchain, toolchain type must be set so that the action executes on the correct execution platform.